lib/PEX.ts

```diff
@@ -304,30 +305,25 @@ export class PEX {
     version?: PEVersion;
     error?: string;
   } {
-    let version = undefined;
-    for (const key of Object.keys(presentationDefinition)) {
-      if (key === 'frame') {
-        if (version === PEVersion.v1) {
-          return { error: 'This is not a valid PresentationDefinition' };
-        }
-        version = PEVersion.v2;
-      } else if (key === 'input_descriptors') {
-        for (const id of presentationDefinition['input_descriptors']) {
-          for (const idKey of Object.keys(id)) {
-            if (idKey === 'schema') {
-              if (version === PEVersion.v2) {
-                return { error: 'This is not a valid PresentationDefinition' };
-              }
-              version = PEVersion.v1;
-            }
-          }
-        }
-      }
-    }
-    if (!version) {
-      version = PEVersion.v2;
+    const v1Validators = [];
+    const v2Validators = [];
+    v1Validators.push({
+      bundler: new PresentationDefinitionV1VB('root'),
+      target: presentationDefinition,
+    });
+    v2Validators.push({
+      bundler: new PresentationDefinitionV2VB('root'),
+      target: presentationDefinition,
+    });
+    const v1Result = new ValidationEngine().validate(v1Validators);
+    const v2Result = new ValidationEngine().validate(v2Validators);
```

Comment:
We are doing 2 validation even if for instance it would have passed as V2.
Given I am pretty confident that we will mostly see V2 in the wild in the future, probably it makes sense to test for that one first. If it fails test for V1

Reply:
changed